### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-dropdown

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1475]
## Description
This task describes the Storybook implementation for the sd-dropdown component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-checkbox can be found here: [sd-dropdown Notes](https://www.notion.so/matuzo/sd-dropdown-112f3f6e1bc080dab9cae2a0771aa8e9).

----------------------------

## Comment
**Problem:** The button in the rounded icon-only example has no accessible name.
**Solution:** Provide one by labeling the icon. => maybe already fixed with the button a11y fixes

**Problem:** The button doesn't indicate whether the dropdown is open.
**Solution:** Put `aria-expanded=false` on the button and toggle its value on click.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
